tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/AdoNet/DbScopeFactory.cs

```diff
@@ -28,14 +28,16 @@ private static Scope CreateDbCommandScope(Tracer tracer, IDbCommand command, Int
             }
 
             Scope scope = null;
+            string commandText = command.CommandText;
 
             try
             {
                 Span parent = tracer.InternalActiveScope?.Span;
 
                 if (parent is { Type: SpanTypes.Sql } &&
                     parent.GetTag(Tags.DbType) == dbType &&
-                    parent.ResourceName == command.CommandText)
+                    (parent.GetTag(Tags.DbmDataPropagated) != null ||
```

Comment:
I don't get this extra check TBH, why do we need it?

Reply:
I don't know this part of the code much. Is there any chance that we reach this code with a parent tag that has been marked as propagated, but a different query?

Reply:
It doesn't seem to be the case based on the example there and the details I found regarding 
[CommandBehavior](https://learn.microsoft.com/en-us/dotnet/api/system.data.commandbehavior?redirectedfrom=MSDN&view=net-7.0), for example after running the query on the DBM these nested methods use the same query so they can:
> Provides a description of the results of the query and its effect on the database.